pkg/commands/workflows/plan_status_comments.go

```diff
@@ -132,99 +109,68 @@ func (c *PlanStatusCommentCommand) Run(ctx context.Context, args []string) error
 		return flag.ErrHelp
 	}
 
-	c.actions = githubactions.New(githubactions.WithWriter(c.Stdout()))
-	actionsCtx, err := c.actions.Context()
+	rc, err := c.createReporterClient(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to load github context: %w", err)
+		return fmt.Errorf("failed to create repoter client: %w", err)
 	}
+	c.reporterClient = rc
 
-	c.cfg = &PlanStatusCommentsConfig{}
-	if err := c.cfg.MapGitHubContext(actionsCtx); err != nil {
-		return fmt.Errorf("failed to load github context: %w", err)
-	}
-	logger.DebugContext(ctx, "loaded configuration", "plan_status_comments_config", c.cfg)
+	return c.Process(ctx)
+}
 
-	tokenSource, err := c.GitHubFlags.TokenSource(ctx, map[string]string{
-		"pull_requests": "write",
-	})
-	if err != nil {
-		return fmt.Errorf("failed to get token source: %w", err)
+// createReporterClient creates a new reporter client based on the reporter flag.
+func (c *PlanStatusCommentCommand) createReporterClient(ctx context.Context) (reporter.Reporter, error) {
```

Comment:
This function has to be duplicated across commands because each reporter type requires a different set of inputs that come from the command. My previous version created an embedded "ReporterCommand" that embeded this function to allow this to be added to any command as needed.